client/src/components/FilterForm.tsx

```diff
@@ -0,0 +1,81 @@
+import { useState } from 'react';
+import { Search as SearchIcon } from 'lucide-react';
+
+type FilterProps = {
+  onSearchInput: (value: string) => void;
+};
+
+export default function FilterForm({ onSearchInput }: FilterProps) {
+  const [selectOption, setSelectOption] = useState({
+    location: '',
+    level: '',
+    salary: '',
+  });
+
+  return (
+    <div>
+      <form>
```

Comment:
There's no "onSubmit". Do you expect to have a submit button? If not, you don't need the `<form>` tag here at all. You can just put buttons anywhere you want if you give them all `onClick` or `onChange` handlers.

Reply:
I don't think I will have a submit button, I was planning on automatically filtering with a debounce. That is good to know I don't need the form wrapper!